scipy/stats/_hypotests.py

```diff
@@ -389,3 +390,192 @@ def _get_wilcoxon_distr(n):
                          "statistic is not implemented for n={}".format(n))
 
     return np.array(cnt, dtype=int)
+
+
+def _pval_cvm_2samp_exact(s, nx, ny):
+    """
+    Compute the exact p-value of the Cramer-von Mises two-sample test
+    for a given value s (float) of the test statistic by enumerating
+    all possible combinations. nx and ny are the sizes of the samples.    
+    """
+    z = np.arange(1, nx+ny+1)
+    rangex = np.arange(1, nx+1)
+    rangey = np.arange(1, ny+1)
+
+    us = []
+    # for the first sample x, select all possible combinations of
+    # the ranks of the elements of x in the combined sample
+    # note that the acutal values of the samples are irrelevant
+    # once the ranks of the elements of x are fixed, the
+    # ranks of the elements in the second sample
+    # that z = 1, ..., nx+ny contains all possible ranks for a loop over all 
+    for c in combinations(z, nx):
+        x = np.array(c)
+        # the ranks of the second sample y are given by the set z - x
+        # note than one needs to shift by -1 since the lowest rank is 1
+        mask = np.ones(nx+ny, bool)
+        mask[x-1] = False
+        y = z[mask]
+        # compute the statistic
+        u = nx * np.sum((x - rangex)**2)
+        u += ny * np.sum((y - rangey)**2)
+        us.append(u)
+    # compute the values of u and the frequencies
+    u, cnt = np.unique(us, return_counts=True)
+    return np.sum(cnt[u >= s]) / np.sum(cnt)
+
+
+def cramervonmises_2samp(x, y, method='auto'):
+    """
+    Perform the two-sample Cramér-von Mises test for goodness of fit.
+
+    This is the two-sample version of the Cramér-von Mises test ([1]_):
+    for two independent samples :math:`X_1, ..., X_n` and
+    :math:`Y_1, ..., Y_m`, the null hypothesis is that the samples
+    come from the same (unspecified) continuous distribution.
+
+    Parameters
+    ----------
+    x : array_like
+        A 1-D array of observed values of the random variables :math:`X_i`.
+    y : array_like
+        A 1-D array of observed values of the random variables :math:`Y_i`.
+    method : {'auto', 'asymp', 'exact'}, optional
+        The method used to compute the p-value, see Notes for details.
+        The default is 'auto'.
+
+    Returns
+    -------
+    res : object with attributes
+        statistic : float
+            Cramér-von Mises statistic.
+        pvalue : float
+            The p-value.
+
+    See Also
+    --------
+    cramervonmises, anderson_ksamp, epps_singleton_2samp, ks_2samp
+
+    Notes
+    -----
+    .. versionadded:: 1.7.0
+
+    The statistic is computed according to equation 9 in [2]_. The
+    calculation of the p-value depends on the keyword `method`:
+
+    - ``asymp``: The p-value is approximated by using the limiting
```

Comment:
```suggestion
    - ``asymptotic``: The p-value is approximated by using the limiting
```
I would suggest the full word as used in [`kendalltau`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.kendalltau.html). That's what I'm using in gh-4933 (mannwhitneyu) and gh-12531 (Page's L), and it would be nice to keep it consistent.

Reply:
changed.